cng/hash.go

```diff
@@ -181,11 +148,11 @@ func hashToID(h hash.Hash) string {
 	return hx.alg.id
 }
 
+// hashX implements [hash.Hash].
 type hashX struct {
-	alg  *hashAlgorithm
-	_ctx bcrypt.HASH_HANDLE // access it using withCtx
```

Comment:
This pattern seems like it makes it easier to spot bad access vs. sprinkling `runtime.KeepAlive`, why the change? (Also IMO would be good for https://github.com/golang-fips/openssl/pull/238#pullrequestreview-2528033583.)

Reply:
It is easy to forget to use `withCtx` (it happened in the OpenSSL module), so it is not a silver bullet. I prefer the `runtime.KeepAlive` pattern because it doesn't add a nested block on each function that access the context (thus the code is simpler), and also to keep consistency with other algorithms, which tend to use the `runtime.KeepAlive` rather than custom `withX` functions.

Reply:
Ok, if withCtx didn't help in practice, I agree that simplifying to `runtime.KeepAlive` makes sense. (If it did help, I would have said that it would make sense to expand that style to other algorithms as the way to get to consistency. 😄)